### ISSUE TITLE
junit reporter reports unit tests twice, when executed via intern-runner.

### ISSUE THREAD

**Issue by bartoszkaczorek** [theintern/intern#365]
When I run unit tests using intern-runner and junit reporter they will be reported twice in the resulting report.xml file. This issue occur only for unit tests, functional tests will be reported only once. This problem also exists when writing a custom reporter and listening to '/suite/end' event, the event will occur twice with the "main" suite.

**Steps to reproduce:**

1. Start selenium server
2. Run tests using command:
 - `node_modules\.bin\intern-runner config=config/intern.js reporters=junit`
3. See that the "report.xml" reports test twice.

**File structure:**
package.json
config/intern.js
exampleTests/exampleTests.js

*package.json*
```json
{
  "name": "intern.junit.reporter.poc",
  "version": "1.0.0",
  "description": "junitReporterPoC",
  "devDependencies": {
    "intern": "~2.2.2"
  }
}
```

*config/intern.js*
```javascript
define([], function () {
	return {
		proxyPort : 9000,
		proxyUrl : 'localhost:9000',
		capabilities : {
			'selenium-version' : '2.43.1'

		},
		environments : [{
			browserName : 'chrome',
			platform : ['WINDOWS']
		}],
		maxConcurrency : 1,
		useSauceConnect : false,
		loader : {
			packages : []
		},
		suites : ['exampleTests/exampleTests'],
		functionalSuites : null,
		excludeInstrumentation : /^exampleTests\//
	};
});
```

*exampleTests.js*
```javascript
define([ 'intern!bdd', 'intern/chai!expect' ], function(bdd, expect) {

	bdd.describe('exampleTests', function() {

		bdd.it('Hello Test', function() {
			var result = 2 + 2;
			expect(result).to.equal(4);
		});

	});

});
```

*Here's the output of the report.xml (test is reported twice):*
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuites>
	<testsuite name="chrome 41.0.2272.89 on XP" failures="0" skipped="0" tests="1" time="0.084">
		<testsuite name="exampleTests" failures="0" skipped="0" tests="1" time="0.039">
			<testcase name="Hello Test" time="0.002" status="0"/>
		</testsuite>
	</testsuite>
	<testsuite name="chrome 41.0.2272.89 on XP" failures="0" skipped="0" tests="1" time="2.142">
		<testsuite name="unit tests" failures="0" skipped="0" tests="1" time="0">
			<testsuite name="main" failures="0" skipped="0" tests="1" time="0.084">
				<testsuite name="exampleTests" failures="0" skipped="0" tests="1" time="0.039">
					<testcase name="Hello Test" time="0.002" status="0"/>
				</testsuite>
			</testsuite>
		</testsuite>
	</testsuite>
</testsuites>
```

**Comment by csnover** [comment]
Thanks for the report, we’ll take a look the next time someone has a chance!

**Comment by csnover** [comment]
This should be fixed when #312 lands.

**Comment by csnover** [comment]
This should be fixed in master today.
